lisa/util/shell.py

```diff
@@ -267,6 +267,14 @@ def _initialize(self, *args: Any, **kwargs: Any) -> None:
         else:
             self.is_posix = True
             shell_type = spur.ssh.ShellTypes.sh
+            # it doesn't support bash. Use minimal shell type
+            if stdout_content and "Unknown syntax" in stdout_content:
+                shell_type = spur.ssh.ShellTypes.minimal
+                func_type = type(spur.ssh.ShellTypes.minimal.generate_run_command)
+                shell_type.generate_run_command = func_type(
```

Comment:
The mypy error is caused by the shell_type is not an instance. So the replacement of generate_run_command should happen after `self._inner_shell = spurplus.SshShell(spur_ssh_shell=spur_ssh_shell, sftp=sftp)`. The `self._inner_shell` is an instance.

Reply:
Thank you Chi. I have added the changes.

Reply:
Thank you Chi. I have added the changes.